### PR TITLE
Overhaul mlflow.metrics module docs

### PR DESCRIPTION
**Pull request by prithvikannan**
### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Update the mlflow.metrics documentation.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/python_api/mlflow.metrics.rst [hunk 1]
```diff
@@ -1,7 +1,58 @@
...
+   :annotation:
+
+.. autodata:: mlflow.metrics.rougeL
+   :annotation:
+
+.. autodata:: mlflow.metrics.rougeLsum
+   :annotation:
+
+.. autodata:: mlflow.metrics.toxicity
+   :annotation:
```
Comment (dbczumar) [inline docs/source/python_api/mlflow.metrics.rst:34]: Can we name / identify the model type(s) for which each of these built-in metrics are automatically computed?

File: docs/source/python_api/mlflow.metrics.rst [hunk 2]
```diff
@@ -1,7 +1,58 @@
 mlflow.metrics
 ==============
 
+The ``mlflow.metrics`` module helps you quantitatively and qualitatively measure your models. These :py:class:`EvaluationMetric <mlflow.metrics.EvaluationMetric>` are used by the :py:func:`mlflow.evaluate()` API, either computed automatically depending on the ``model_type`` or specified via the ``custom_metrics`` parameter. Evaluation results are stored as :py:class:`MetricValue <mlflow.metrics.MetricValue>` and are logged to the MLflow run.
```
Comment (dbczumar) [inline docs/source/python_api/mlflow.metrics.rst:4]: Can we add code example(s) demonstrating how to use `mlflow.evaluate()` with the intelligent metrics?

**Review by dbczumar — APPROVED** [review]
LGTM with tiny comments - thanks @prithvikannan  - looks great!

File: docs/source/python_api/mlflow.metrics.rst (hunk 1 above)
Comment (prithvikannan) [reply docs/source/python_api/mlflow.metrics.rst:67]: Will address in a followup PR once the regression and classification metrics are also merged.

**Pull request merged**